src/test/java/io/xstefank/wildfly/bot/PushEventTest.java

```diff
@@ -0,0 +1,224 @@
+                    Thread.sleep(sleepTimeMilis);
+                    verify(mocks.repository(TEST_REPO)).queryPullRequests();
+                    for (int i = 0; i < pullRequestCount; i++) {
+                        Mockito.verify(mocks.pullRequest(i), times(5)).getMergeable();
+                        verify(mocks.pullRequest(i)).getLabels();
+                        if (i % 2 == 0 && i >= 2) {
+                            verify(mocks.pullRequest(i)).addLabels(LABEL_NEEDS_REBASE);
+                        }
+                        if (i % 2 == 1 && i < 2) {
+                            verify(mocks.pullRequest(i)).removeLabels(LABEL_NEEDS_REBASE);
+                        }
+                        verifyNoMoreInteractions(mocks.pullRequest(i));
+                    }
+                    // Same as above... This happens as we mock repository content and on second mocking we invoke the mocked method...
+                    verify(mocks.repository(TEST_REPO)).getDirectoryContent(anyString());
+                    Mockito.verifyNoMoreInteractions(mocks.repository(TEST_REPO));
+                });
+    }
+
+    /**
+     * We test that a queue is created and the Push Events are run reactively
+     */
+    @Test
+    void creatingQueueOnPushEventTest() throws Exception {
+        given().github(mocks -> {
+            GHRepository repository = mocks.repository(TEST_REPO);
+            List<GHPullRequest> pullRequests = new ArrayList<>();
+            GHPullRequest mockedPR = mocks.pullRequest(123);
+            when(mockedPR.getMergeable()).thenReturn(null).thenAnswer(new Answer<>() {
+                private Instant validAfter;
+
+                @Override
+                public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                    if (validAfter == null) {
+                        validAfter = Instant.now().plusSeconds(wildFlyBotConfig.timeout());
+                        return null;
+                    }
+
+                    if (validAfter.isAfter(Instant.now())) {
+                        return Boolean.TRUE;
+                    }
+                    return null;
+                }
+            });
+            pullRequests.add(mockedPR);
+            GHPullRequestQueryBuilder mockedQueryBuilder = Mockito.mock(GHPullRequestQueryBuilder.class);
+            PagedIterable<GHPullRequest> pullRequestsPagedIterable = Mockito.mock(PagedIterable.class);
+
+            when(repository.queryPullRequests()).thenReturn(mockedQueryBuilder);
+
+            when(mockedQueryBuilder.base(anyString())).thenReturn(mockedQueryBuilder);
+            when(mockedQueryBuilder.state(GHIssueState.OPEN)).thenReturn(mockedQueryBuilder);
+            when(mockedQueryBuilder.list()).thenReturn(pullRequestsPagedIterable);
+
+            when(pullRequestsPagedIterable.toList()).thenReturn(pullRequests);
+        })
+                .when().payloadFromClasspath(PAYLOAD).event(GHEvent.PUSH)
+                .then();
```

Comment:
missing verifiaction?

Reply:
This is done intentionally. We want to create 2 PushEvents and verify (Be it only with logs), that there is only one executed Uni at a time, i.e. the other is queued.